File: joyal670-protenium-iro-hub-9d65cb9ae2c4/app/src/main/java/com/protenium/irohub/model/menu/MealPlanLimit.java

package com.protenium.irohub.model.menu;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;


@SuppressWarnings("unused")
public class MealPlanLimit implements Parcelable {

    @SerializedName("meal_type_id")
    private String mMealTypeId;
    @SerializedName("quantity")
    private String mQuantity;

    protected MealPlanLimit(Parcel in) {
        mMealTypeId = in.readString();
        mQuantity = in.readString();
    }

    public static final Creator<MealPlanLimit> CREATOR = new Creator<MealPlanLimit>() {
        @Override
        public MealPlanLimit createFromParcel(Parcel in) {
            return new MealPlanLimit(in);
        }

        @Override
        public MealPlanLimit[] newArray(int size) {
            return new MealPlanLimit[size];
        }
    };

    public String getMealTypeId() {
        return mMealTypeId;
    }

    public void setMealTypeId(String mealTypeId) {
        mMealTypeId = mealTypeId;
    }

    public String getQuantity() {
        return mQuantity;
    }

    public void setQuantity(String quantity) {
        mQuantity = quantity;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(mMealTypeId);
        dest.writeString(mQuantity);
    }
}

File: proteinium/app/src/main/java/com/proteinium/proteiniumdietapp/pojo/menu_day/MealPlanLimit.kt
package com.proteinium.proteiniumdietapp.pojo.menu_day

import android.os.Parcel
import android.os.Parcelable

data class MealPlanLimit(
    val meal_type_id: String,
    val quantity: String
):Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString().toString(),
        parcel.readString().toString()
    ) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(meal_type_id)
        parcel.writeString(quantity)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MealPlanLimit> {
        override fun createFromParcel(parcel: Parcel): MealPlanLimit {
            return MealPlanLimit(parcel)
        }

        override fun newArray(size: Int): Array<MealPlanLimit?> {
            return arrayOfNulls(size)
        }
    }
}